src/App.tsx

```diff
@@ -16,35 +22,96 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const handleReverse = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const handleReset = () => {
+    setSortField(SortType.DEFAULT);
+    setIsReversed(false);
+  };
+
+  const getSortedGoods = () => {
+    const sortedGoods = [...goodsFromServer];
+
+    if (sortField === SortType.ALPHABETICALLY) {
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+    } else if (sortField === SortType.BY_LENGTH) {
+      sortedGoods.sort((a, b) => a.length - b.length);
+    }
+
+    if (isReversed) {
+      sortedGoods.reverse();
+    }
+
+    return sortedGoods;
+  };
+
+  const visibleGoods = getSortedGoods();
+
+  const isOriginalOrder = sortField === SortType.DEFAULT && !isReversed;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={
+            sortField === SortType.ALPHABETICALLY
+              ? 'button is-info'
+              : 'button is-info is-light'
+          }
+          onClick={() => {
+            setSortField(SortType.ALPHABETICALLY);
+          }}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={
+            sortField === SortType.BY_LENGTH
+              ? 'button is-info'
+              : 'button is-info is-light'
+          }
+          onClick={() => {
+            setSortField(SortType.BY_LENGTH);
+          }}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={
+            isReversed ? 'button is-warning' : 'button is-info is-light'
+          }
+          onClick={handleReverse}
```

Comment:
[UX] - don't forget to mark the active button to explain to the user what action is active.